### PR TITLE
feat(codex): show reasoning effort in usage logs

### PR DESCRIPTION
**Pull request by baitaotao521**
## Summary

- record Codex Responses API `reasoning.effort` in usage-log audit data
- show an always-visible reasoning-effort column immediately after the billing model in both regular and virtualized usage-log tables
- display requested and effective effort values when a provider override changes the request
- add the official `max` effort tier to provider types, validation, batch patches, forms, and all five locales
- keep `ultra` invalid because it is a Codex product mode, not a Responses API `reasoning.effort` value
- generalize the existing effort badge so Anthropic and Codex levels share consistent rendering

## Motivation

The current OpenAI Responses API supports a new `max` reasoning-effort tier. Codex usage records previously did not persist or surface `reasoning.effort`, which made it difficult to verify the value actually forwarded after provider-level overrides.

Official references:

- https://developers.openai.com/api/reference/resources/responses/methods/create/
- https://developers.openai.com/codex/models/

## Related Work

- Follow-up to #536 — Codex provider-level `reasoning`/`text`/`parallel_tool_calls` overrides. That PR introduced the `codex_reasoning_effort_preference` field and the `provider_parameter_override` audit. This PR records the client's requested `reasoning.effort` into `special_settings` before the usage record is created, renders the effective forwarded value from that same `provider_parameter_override` audit, and adds the official `max` tier to the preference field #536 created.
- Related to #1026 — added the `xhigh` tier for Anthropic adaptive thinking. This PR follows the same type (`src/types/provider.ts`) / Zod validation (`src/lib/validation/schemas.ts`) / batch-patch contract (`src/lib/provider-patch-contract.ts`) pattern to add `max` for Codex reasoning effort, and generalizes the shared badge (`AnthropicEffortBadge` -> `ThinkingEffortBadge`) so Anthropic and Codex levels render with consistent styling.
- Related to #1269 — thinking effort conflict rectifier. Both extend the `special-settings` audit type union (`src/types/special-settings.ts`, `src/lib/utils/special-settings.ts`) that this PR also adds the `codex_reasoning_effort` audit type to.

## Behavior

- Codex requests with `reasoning.effort` persist the original request value in `special_settings`.
- Provider override audit data remains the source of the effective forwarded value.
- Usage logs render `requested -> effective` when the provider changes the value.
- Historical records containing only provider override audit data remain displayable.
- Records without Codex reasoning-effort audit data render a neutral placeholder.

### Core Changes
- `src/app/v1/_lib/proxy/message-service.ts` — extract `reasoning.effort` and write a `codex_reasoning_effort` audit entry into `special_settings` before the usage record is created (Codex providers only; deduped; skips when absent).
- `src/lib/utils/codex-reasoning-effort.ts` (new) — request-body extraction, special-settings extraction, and `extractCodexReasoningEffortInfo` which merges the requested value with the `provider_parameter_override` effective value (including backward compatibility with override-only historical records).
- `src/types/special-settings.ts`, `src/lib/utils/special-settings.ts` — new `codex_reasoning_effort` audit type + dedup key.
- `src/components/customs/thinking-effort-badge.tsx` — renamed from `anthropic-effort-badge.tsx`; `AnthropicEffortBadge` -> `ThinkingEffortBadge`, adds `none`/`minimal` styles so Anthropic and Codex levels share one renderer.

### Usage-log UI
- `usage-logs-table.tsx`, `virtualized-logs-table.tsx` — new always-visible "Reasoning Effort" column immediately after the billing model; empty-state `colSpan` bumped to 13; virtualized `min-w` 800px -> 900px.
- `src/app/[locale]/dashboard/logs/_components/codex-reasoning-effort-display.tsx` (new) — renders requested value, and `requested -> effective` with an arrow + "overridden by provider" note when a provider override changed it.
- `src/lib/column-visibility.ts` — add `reasoningEffort` to `ALWAYS_VISIBLE_COLUMNS`.
- `error-details-dialog/components/SummaryTab.tsx` — unified `effortDisplay` (Codex reasoning effort preferred, else Anthropic effort) using `ThinkingEffortBadge`.

### `max` effort tier
- `src/types/provider.ts`, `src/lib/validation/schemas.ts`, `src/lib/provider-patch-contract.ts` — add `max` to `CodexReasoningEffortPreference`; `ultra` stays invalid (Codex product mode, not a Responses API value).
- `options-section.tsx` — reorder options and include `max`.
- All 5 locale `settings/providers/form/sections.json` — `max` option + generalized help text (model-dependent levels).

## Breaking Changes

None. No database migration (the new audit lives in the existing `special_settings` JSON column). The badge component rename is repo-internal and all call sites were updated in this PR.

## Validation

- [x] `bun run build`
- [x] `bun run lint`
- [x] `bun run lint:fix`
- [x] `bun run typecheck`
- [x] 354 focused unit/component tests
- [x] Docker image build and local runtime health checks
- [x] `bun run i18n:audit-messages-no-emoji:fail`
- [x] i18n settings split and module-load tests
- [x] placeholder audit completed; existing candidates only: `ja=9`, `zh-TW=60`
- [x] no placeholder allowlist changes

Full local suite result: 6751 passed, 4 unrelated pre-existing/local-environment failures:

- macOS locale causes `tr: Illegal byte sequence` in the Kubernetes shell-helper fallback test
- two user-insights date assertions differ by one day because of the local timezone
- the blocked-sessionStorage language-switcher test does not observe its expected console error

None of those failing files are changed by this PR.

## i18n audit summary

| Locale | Settings candidates | Dashboard candidates | My Usage candidates |
| --- | ---: | ---: | ---: |
| ja | 7 | 2 | 0 |
| zh-TW | 45 | 15 | 0 |

The new and modified strings are translated for `zh-CN`, `zh-TW`, `en`, `ja`, and `ru`. No emoji or new placeholder candidates were introduced.

### Automated Tests
- [x] Unit tests added/updated — `codex-reasoning-effort.test.ts`, `message-service.test.ts`, `thinking-effort-badge.test.ts`, `schemas.test.ts`, `column-visibility.test.ts`, plus table/dialog component tests and `codex-provider-overrides.test.ts` / `providers-patch-contract.test.ts` cases for `max`.

### Manual Testing
1. Create or edit a Codex provider and set the reasoning-effort override to `max`; confirm `max` appears and is persisted.
2. Send a Codex request with `reasoning.effort`; confirm the new column shows the requested value, and `requested -> effective` when a provider override changes it.
3. Open a non-Codex log and confirm the neutral `-` placeholder renders.

## Checklist

- [x] branch created from the latest `origin/dev`
- [x] PR targets `dev`
- [x] Conventional Commit used
- [x] no database migration required
- [x] no unrelated source changes

---
*Description enhanced by Claude AI*

<details open><summary><h3>Greptile Summary</h3></summary>

This PR adds end-to-end visibility of Codex `reasoning.effort` in usage logs: the proxy now captures the client's requested value into `special_settings` before the usage record is created, and a new always-visible table column shows "requested → effective" when a provider-level override changes the value. It also adds the official `max` effort tier and generalizes the existing `AnthropicEffortBadge` into a shared `ThinkingEffortBadge`.

- **Audit capture** (`message-service.ts`, `codex-reasoning-effort.ts`): a new `codex_reasoning_effort` special-setting is written for Codex providers before the record is flushed, giving the display layer a stable "requested" anchor alongside the existing `provider_parameter_override` "effective" anchor.
- **Usage-log UI** (`usage-logs-table.tsx`, `virtualized-logs-table.tsx`, `codex-reasoning-effort-display.tsx`): a new "Reasoning Effort" column renders the badge, arrow, and override note; backward-compatible with historical records that only carry provider-override audit data.
- **`max` tier + badge generalization** (`provider.ts`, `schemas.ts`, `thinking-effort-badge.tsx`, all 5 locales): `max` is added to the Codex preference enum and the shared badge now covers all levels from `none` through `max`, with `ultra` correctly kept out of scope.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — all changed paths are additive; no existing data is modified and the new audit type writes to the existing JSON column.

The core extraction logic correctly handles all edge cases (null efforts, backward-compat historical records, multi-provider retries) and is covered by focused unit tests. The table column, badge generalisation, and `max` tier additions are self-contained and do not alter any existing behavior. The typecheck and full local test suite both pass.

No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/lib/utils/codex-reasoning-effort.ts | New utility with three well-guarded extraction functions. The multi-override accumulation logic (first before, last after) is correct for both the new-record and historical backward-compat paths. All edge cases—null efforts, non-string values, empty strings, missing audit entries—are handled via `normalizeCodexReasoningEffort`. |
| src/app/v1/_lib/proxy/message-service.ts | Adds Codex reasoning-effort audit write before `createMessageRequest`, mirroring the existing Anthropic effort pattern. The `!hasCodexReasoningEffortAudit` guard correctly makes the write idempotent across retries. |
| src/app/[locale]/dashboard/logs/_components/codex-reasoning-effort-display.tsx | Clean display component: null-guards on `requestedEffort` correctly handle the backward-compat historical-record path; tooltip shows 'Overridden by provider' only when `isOverridden` is true. |
| src/components/customs/thinking-effort-badge.tsx | Renamed from `anthropic-effort-badge.tsx`; adds `none`/`minimal` styles and `max` style; unknown levels fall back to a neutral default style. Visual ordering none < minimal < low < medium < high < xhigh < max is reflected in color progression. |
| src/app/[locale]/dashboard/logs/_components/error-details-dialog/components/SummaryTab.tsx | Unified `effortDisplay` correctly prefers Codex reasoning effort over Anthropic effort; null-safe rendering of `effectiveEffort` (which is `string | null` for the Anthropic path) is guarded by `effortDisplay.isOverridden && effortDisplay.effectiveEffort`. |
| src/types/special-settings.ts | Adds `CodexReasoningEffortSpecialSetting` type to the discriminated union; dedup key in `special-settings.ts` is consistent with other audit types. |
| src/types/provider.ts | Adds `max` to `CodexReasoningEffortPreference`; `ultra` is correctly excluded per the PR rationale. |
| src/lib/validation/schemas.ts | Adds `max` to the Zod enum for `CODEX_REASONING_EFFORT_PREFERENCE`; consistent with the type definition update. |
| src/app/[locale]/dashboard/logs/_components/usage-logs-table.tsx | New 'Reasoning Effort' column added after 'Billing Model'; empty-state `colSpan` correctly updated to 13 to match the new column count. |
| src/app/[locale]/dashboard/logs/_components/virtualized-logs-table.tsx | New Reasoning Effort column added in the virtualized table; `min-w` updated from 800px to 900px to accommodate the new column width. |

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant Client
    participant ProxyMessageService
    participant Session
    participant DB
    participant UI

    Client->>ProxyMessageService: Codex request (reasoning.effort: "high")
    ProxyMessageService->>Session: extractCodexReasoningEffortFromRequestBody()
    Session-->>ProxyMessageService: "high"
    ProxyMessageService->>Session: "addSpecialSetting({type: codex_reasoning_effort, effort: high})"
    ProxyMessageService->>DB: "createMessageRequest(special_settings=[codex_reasoning_effort])"

    note over Session,DB: Provider override runs after record creation
    Session->>DB: "update special_settings += provider_parameter_override(before:high, after:max)"

    UI->>DB: fetch usage log row
    DB-->>UI: specialSettings with codex_reasoning_effort + provider_parameter_override
    UI->>UI: extractCodexReasoningEffortInfo(specialSettings)
    note over UI: requestedEffort=high, effectiveEffort=max, isOverridden=true
    UI->>UI: render high → max with Overridden by provider
```
</details>

<sub>Reviews (5): Last reviewed commit: ["chore: format Codex reasoning tests"](https://github.com/ding113/claude-code-hub/commit/68804cc321ff12fe90d1118e53b3fe1a0f9fbc2b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=43119113)</sub>

### CODE REVIEW

File: src/lib/utils/codex-reasoning-effort.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  let overrideBefore: string | null = null;
+  let overrideAfter: string | null = null;
+  let overrideChanged = false;
+
+  for (const setting of specialSettings) {
+    if (setting.type !== "provider_parameter_override" || setting.providerType !== "codex") {
+      continue;
+    }
+
+    const effortChange = setting.changes.find((change) => change.path === "reasoning.effort");
```
[the remark this answers is not in the document]
Comment (baitaotao521) [reply src/lib/utils/codex-reasoning-effort.ts]: Addressed in 3a746f12. The extractor now uses optional chaining and skips malformed historical audits, with regression coverage for a missing changes array.

File: src/app/[locale]/dashboard/logs/_components/usage-logs-table.tsx [hunk 2]
```diff
@@ -297,6 +299,9 @@ export function UsageLogsTable({
                         </Tooltip>
                       </TooltipProvider>
                     </TableCell>
+                    <TableCell className="font-mono text-xs w-[120px] max-w-[120px]">
+                      <CodexReasoningEffortDisplay specialSettings={log.specialSettings} />
+                    </TableCell>
```
[the remark this answers is not in the document]
Comment (baitaotao521) [reply src/app/[locale]/dashboard/logs/_components/usage-logs-table.tsx]: Addressed in 3a746f12. The constrained table cell now clips overflow, and the override rendering test verifies both effort values and the overflow guard.

File: src/lib/column-visibility.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export const DEFAULT_HIDDEN_COLUMNS: LogsTableColumn[] = ["ip"];
 /**
  * Columns that cannot be hidden (always visible)
  */
-export const ALWAYS_VISIBLE_COLUMNS = ["time", "model", "status"] as const;
+export const ALWAYS_VISIBLE_COLUMNS = ["time", "model", "reasoningEffort", "status"] as const;
```
[the remark this answers is not in the document]
Comment (baitaotao521) [reply src/lib/column-visibility.ts:47]: Confirmed that ALWAYS_VISIBLE_COLUMNS is an existing documentation constant rather than the production rendering gate. All fixed columns are rendered directly and omitted from the toggleable set. Refactoring that legacy convention is outside this feature, and the reasoning-effort column is already wired in both table implementations.
